tools/crm_error.c

```diff
@@ -124,9 +122,10 @@ main(int argc, char **argv)
             if (pcmk__str_eq(name, "Unknown", pcmk__str_null_matches) || !strcmp(name, "CRM_EX_UNKNOWN")) {
                 // Undefined
             } else if(options.with_name) {
-                printf("% .*d: %-26s  %s\n", width, rc, name, desc);
+                // Room for four digits and a negative sign, space-padded
+                printf("% 5d: %-30s  %s\n", rc, name, desc);
```

Comment:
Rather than expand the fields, I think we should loop once through all the values and find the longest name and value, and use the relevant widths.

Reply:
This will be easier if we wait for until the end of the commits in this PR, rather than doing it in `crm_error.c` early on. Later, I add an internal constant `pcmk__n_rc` that exposes the number of negative return codes outside of `results.c`. It replaces `PCMK__N_RC`.

I think if we're going to get name and value widths dynamically, it only makes sense to base it on the standard Pacemaker return codes. In practice, the widest name and value of that class should always be wider than the widest of any other class. I'll do this in the next update.

The alternative is to add one or more "width" arguments to the formatter functions and let the caller loop over the relevant values to determine the width(s) it needs. I don't like that, because (1) those arguments would be ignored by the XML format function., and (2) less importantly, it would add one or two extra params to deal with.

Reply:
I just moved all the refactor and non-API feature stuff to the beginning of the commit chain and did them in `crm_error.c` instead of in `pcmk_result_code.c` and `pcmk_output.c`, for easier review.